perllib/OIDC/Lite/Client/WebServer/Azure.pm

```diff
@@ -0,0 +1,47 @@
+package OIDC::Lite::Client::WebServer::Azure;
+
+use strict;
+use warnings;
+use parent 'OIDC::Lite::Client::WebServer';
+
+use OIDC::Lite::Client::IDTokenResponseParser;
+
+=head1 NAME
+
+OIDC::Lite::Client::WebServer::Azure - extension to auth against Azure AD B2C
+
+OIDC::Lite doesn't appear to support the authorisation code flow to get an
+ID token - only an access token. Azure returns all its claims in the id_token
+and doesn't support a UserInfo endpoint, so this extension adds a new
+get_id_token method that fetches the ID token.
+
+=cut
+
+=head2 get_id_token( %params )
+
+Fetches and returns L<OIDC::Lite::Model::IDToken> object.
+
+Parameters as the same as for get_access_token, which this method is a small
+wrapper around.
+
+NB this does not perform any verification of the id_token. It's assumed to be
+safe as it's come directly from the OpenID IdP and not an untrusted user's
+browser.
+
+=cut
+
+
+sub get_id_token {
+    my $self = shift;
+
+    my $old_parser = $self->{response_parser};
+
+    $self->{response_parser} = OIDC::Lite::Client::IDTokenResponseParser->new;
```

Comment:
This looks fine, but is there any reason you don't set the parser once in a `new` function; it looks like it's only refresh_access_token is other fn that uses it and presumably that's not called.

Reply:
That's a better idea, thanks.